sdk/remoterendering/mixedreality-remoterendering/src/authentication/mixedRealityTokenCredential.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-auth";
+
+import {
+  MixedRealityStsClient,
+  MixedRealityStsClientOptions
+} from "@azure/mixedreality-authentication";
+import { StaticAccessTokenCredential } from "./staticAccessTokenCredential";
+
+/**
+ * Represents a token credential that can be used to access a Mixed Reality service.
+ * @internal
+ */
+export class MixedRealityTokenCredential implements TokenCredential {
+  private stsClient: MixedRealityStsClient;
+
+  constructor(
+    accountId: string,
+    accountDomain: string,
+    credential: TokenCredential,
+    options: MixedRealityStsClientOptions
+  ) {
+    this.stsClient = new MixedRealityStsClient(accountId, accountDomain, credential, options);
+  }
+
+  getToken(_scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null> {
+    return this.stsClient.getToken(options);
+  }
+
+  static getMixedRealityCredential(
```

Comment:
if you're going to us a static helper like this, you need to mark the constructor private, but at that point, why export the class at all? Why not have this module export a function that returns a `TokenCredential`?

Reply:
I've refactored things so the class is constructed without using this static method.